indexer/base/kind.go

```diff
@@ -0,0 +1,309 @@
+package indexerbase
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// Kind represents the basic type of a column in the table schema.
+// Each kind defines the types of go values which should be accepted
+// by listeners and generated by decoders when providing entity updates.
+type Kind int
+
+const (
+	// InvalidKind indicates that an invalid type.
+	InvalidKind Kind = iota
+
+	// StringKind is a string type and values of this type must be of the go type string
+	// or implement fmt.Stringer().
+	StringKind
+
+	// BytesKind is a bytes type and values of this type must be of the go type []byte.
+	BytesKind
+
+	// Int8Kind is an int8 type and values of this type must be of the go type int8.
+	Int8Kind
+
+	// Uint8Kind is a uint8 type and values of this type must be of the go type uint8.
+	Uint8Kind
+
+	// Int16Kind is an int16 type and values of this type must be of the go type int16.
+	Int16Kind
+
+	// Uint16Kind is a uint16 type and values of this type must be of the go type uint16.
+	Uint16Kind
+
+	// Int32Kind is an int32 type and values of this type must be of the go type int32.
+	Int32Kind
+
+	// Uint32Kind is a uint32 type and values of this type must be of the go type uint32.
+	Uint32Kind
+
+	// Int64Kind is an int64 type and values of this type must be of the go type int64.
+	Int64Kind
+
+	// Uint64Kind is a uint64 type and values of this type must be of the go type uint64.
+	Uint64Kind
+
+	// IntegerKind represents an arbitrary precision integer number. Values of this type must
+	// be of the go type int64, string or a type that implements fmt.Stringer with the resulted string
+	// formatted as an integer number.
+	IntegerKind
+
+	// DecimalKind represents an arbitrary precision decimal or integer number. Values of this type
+	// must be of the go type string or a type that implements fmt.Stringer with the resulting string
+	// formatted as decimal numbers with an optional fractional part. Exponential E-notation
+	// is supported but NaN and Infinity are not.
+	DecimalKind
+
+	// BoolKind is a boolean type and values of this type must be of the go type bool.
+	BoolKind
+
+	// TimeKind is a time type and values of this type must be of the go type time.Time.
+	TimeKind
+
+	// DurationKind is a duration type and values of this type must be of the go type time.Duration.
+	DurationKind
+
+	// Float32Kind is a float32 type and values of this type must be of the go type float32.
```

Comment:
wondering if we want to support this day0 considering they're not used in the state machine

Reply:
I guess I'm thinking better to be thorough, but I don't have a strong opinion